packages/mds-metrics-service/service/repository/index.ts

```diff
@@ -29,59 +23,54 @@ import { ReadMetricsOptions, MetricDomainModel } from '../../@types'
 import * as migrations from './migrations'
 import { MetricEntityToDomain, MetricDomainToEntityCreate } from './mappers'
 
-const RepositoryReadMetrics = CreateRepositoryMethod(connect => async (options: ReadMetricsOptions): Promise<
-  MetricDomainModel[]
-> => {
-  try {
-    const { name, time_bin_size, time_bin_start, time_bin_end, provider_id, geography_id, vehicle_type } = options
-    const connection = await connect('ro')
-    const entities = await connection.getRepository(MetricEntity).find({
-      where: {
-        name,
-        time_bin_size,
-        time_bin_start: Between(
-          timeframe(time_bin_size, time_bin_start).start_time,
-          timeframe(time_bin_size, time_bin_end ?? time_bin_start).end_time
-        ),
-        ...entityPropertyFilter<MetricEntity, 'provider_id'>('provider_id', provider_id),
-        ...entityPropertyFilter<MetricEntity, 'geography_id'>('geography_id', geography_id),
-        ...entityPropertyFilter<MetricEntity, 'vehicle_type'>('vehicle_type', vehicle_type)
-      }
-    })
-    return entities.map(MetricEntityToDomain.mapper())
-  } catch (error) {
-    throw RepositoryError(error)
+class MetricsReadWriteRepository extends ReadWriteRepository {
+  public readMetrics = async (options: ReadMetricsOptions): Promise<MetricDomainModel[]> => {
+    const { connect } = this
+    try {
+      const { name, time_bin_size, time_bin_start, time_bin_end, provider_id, geography_id, vehicle_type } = options
+      const connection = await connect('ro')
+      const entities = await connection.getRepository(MetricEntity).find({
+        where: {
+          name,
+          time_bin_size,
+          time_bin_start: Between(
+            timeframe(time_bin_size, time_bin_start).start_time,
+            timeframe(time_bin_size, time_bin_end ?? time_bin_start).end_time
+          ),
+          ...entityPropertyFilter<MetricEntity, 'provider_id'>('provider_id', provider_id),
+          ...entityPropertyFilter<MetricEntity, 'geography_id'>('geography_id', geography_id),
+          ...entityPropertyFilter<MetricEntity, 'vehicle_type'>('vehicle_type', vehicle_type)
+        }
+      })
+      return entities.map(MetricEntityToDomain.mapper())
+    } catch (error) {
+      throw RepositoryError(error)
+    }
   }
-})
 
-const RepositoryWriteMetrics = CreateRepositoryMethod(connect => async (metrics: MetricDomainModel[]): Promise<
-  MetricDomainModel[]
-> => {
-  try {
-    const connection = await connect('rw')
-    const { raw: entities }: InsertReturning<MetricEntity> = await connection
-      .getRepository(MetricEntity)
-      .createQueryBuilder()
-      .insert()
-      .values(metrics.map(MetricDomainToEntityCreate.mapper()))
-      .returning('*')
-      .execute()
-    return entities
-  } catch (error) {
-    throw RepositoryError(error)
+  public writeMetrics = async (metrics: MetricDomainModel[]): Promise<MetricDomainModel[]> => {
+    const { connect } = this
```

Comment:
In my experience `this` can get a bit wonky wrt. typescript

Reply:
Mine too. That's why class methods are specifically implemented using `pubic method = () => {}` so that the _this_ binding is maintained consistently.